### PR TITLE
[RISC-V] Implement emulate single step feature.

### PR DESCRIPTION
**Pull request by viewizard**
With this changes managed debugger (tested with netcoredbg) can continue execution after break on managed breakpoint.
Note, `RiscV64SingleStepper` code is close to `Arm64SingleStepper` code except arch-related methods like `TryEmulate()`, `GetReg()`, `SetReg()`.

CC @clamp03 @wscho77 @HJLeee @JongHeonChoi @t-mustafin @gbalykov @brucehoult

### CODE REVIEW

**Review by t-mustafin — APPROVED** [review]

File: src/coreclr/vm/riscv64/singlestepper.cpp [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+        fEmulated = true;
+        if (execute)
+        {
+            LOG((LF_CORDB, LL_INFO100000, "RiscV64SingleStepper::TryEmulate JAL\n"));
+
+            // J-immediate encodes a signed offset in multiples of 2 bytes
+            //      21       | 20                                               1 | 0
+            // inst[31]/sign | inst[19:12] | inst[20] | inst[30:25] | inst[24:21] | 0
+            uint64_t imm = SignExtend((BitExtract(opcode, 30, 21) << 1) | (BitExtract(opcode, 20, 20) << 11) |
+                                      (BitExtract(opcode, 19, 12) << 12) | (BitExtract(opcode, 31, 31) << 20), 21);
```
Comment (clamp03) [inline src/coreclr/vm/riscv64/singlestepper.cpp:379]: I think sign bit (inst[31] << 20) in inst is checked in SignExtend with (1ull << 20) not 21. Could you check it is correct?

File: src/coreclr/vm/riscv64/singlestepper.cpp [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+            if ((((opcode & 0x707f) == 0x63) && Rs1SValue == Rs2SValue) ||
+                (((opcode & 0x707f) == 0x1063) && Rs1SValue != Rs2SValue) ||
+                (((opcode & 0x707f) == 0x4063) && Rs1SValue < Rs2SValue) ||
+                (((opcode & 0x707f) == 0x5063) && Rs1SValue >= Rs2SValue))
+            {
+                // B-immediate encodes a signed offset in multiples of 2 bytes
+                //       13      | 12                               1 | 0
+                // inst[31]/sign | inst[7] | inst[30:25] | inst[11:8] | 0
+                uint64_t imm = SignExtend((BitExtract(opcode, 11, 8) << 1) | (BitExtract(opcode, 30, 25) << 5) |
+                                          (BitExtract(opcode, 7, 7) << 11) | (BitExtract(opcode, 31, 31) << 12), 13);
```
Comment (clamp03) [inline src/coreclr/vm/riscv64/singlestepper.cpp:428]: Please check too.

File: src/coreclr/vm/riscv64/singlestepper.cpp [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+            uint64_t Rs2Value = GetReg(pCtx, Rs2);
+
+            if ((((opcode & 0x707f) == 0x6063) && Rs1Value < Rs2Value) ||
+                (((opcode & 0x707f) == 0x7063) && Rs1Value >= Rs2Value))
+            {
+                // B-immediate encodes a signed offset in multiples of 2 bytes
+                //       13      | 12                               1 | 0
+                // inst[31]/sign | inst[7] | inst[30:25] | inst[11:8] | 0
+                uint64_t imm = SignExtend((BitExtract(opcode, 11, 8) << 1) | (BitExtract(opcode, 30, 25) << 5) |
+                                          (BitExtract(opcode, 7, 7) << 11) | (BitExtract(opcode, 31, 31) << 12), 13);
```
Comment (clamp03) [inline src/coreclr/vm/riscv64/singlestepper.cpp:455]: Please check this too.

File: src/coreclr/vm/riscv64/singlestepper.cpp (hunk 1 above)
Comment (viewizard) [reply src/coreclr/vm/riscv64/singlestepper.cpp:379]: Thanks! I miss count bits when write comments and just set wrong sign bit even didn't looked at real sign bit that I moved properly -_- , should be
```
            //      20       | 19                                               1 | 0
            // inst[31]/sign | inst[19:12] | inst[20] | inst[30:25] | inst[24:21] | 0
```
will fix here and other lines too.

**Review by clamp03 — APPROVED** [review]

**Review by jkotas — APPROVED** [review]
Thanks

**Comment by gbalykov** [comment]
@jkotas can you please add @viewizard to .net org?

**Comment by gbalykov** [comment]
> Why is it needed?

He's also responsible for Tizen, specifically for debugging stuff. I though that I already added him to tizen area owners, but it was not the case, so here's a PR: https://github.com/dotnet/runtime/pull/94778.

**Comment by jkotas** [comment]
I meant - are there extra github permissions that you expect to get from dotnet org membership? What are members allowed to do that non-members cannot?

**Comment by gbalykov** [comment]
> I meant - are there extra github permissions that you expect to get from dotnet org membership? What are members allowed to do that non-members cannot?

Nothing extra. If it's the same as for non-members, then there's probably no need to add new people.

**Review by gbalykov — COMMENTED** [review]
LGTM

**Pull request merged**